rpi_collector/synced_frame_builder.py

```
@@ -0,0 +1,407 @@
+        self.previous_sensor_ns: Optional[Ns] = None
+        self.frame_seq = 1
+        self.last_prune_ns = time.monotonic_ns()
+
+    def ingest(self, enriched: Dict[str, Any]) -> None:
+        topic = enriched.get("mqttTopic")
+        with self.lock:
+            if topic == TOPIC_CSI_RAW:
+                if csi_received_ns(enriched) is not None:
+                    self.csi_buffer.append(enriched)
+            elif topic == TOPIC_SENSOR_RAW:
+                if sensor_received_ns(enriched) is not None:
```

Ruff (0.15.15)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)